pyquil/api/_quantum_computer.py

```diff
@@ -154,40 +131,87 @@ def run(self, executable: Executable,
             .read_memory(region_name='ro')
 
     @_record_call
-    def run_symmetrized_readout(self, program: Program, trials: int) -> np.ndarray:
+    def run_symmetrized_readout(self, program: Program, trials: int, symm_type: int = 3,
+                                    meas_qubits: List[int] = None) -> np.ndarray:
+        r"""
+        Run a quil program in such a way that the readout error is made symmetric. Enforcing
+        symmetric readout error is useful in simplifying the assumptions in some near
+        term error mitigation strategies, see `measure_observables` for more information.
+
+        The simplest example is for one qubit. In a noisy device, the probability of accurately
+        reading the 0 state might be higher than that of the 1 state; due to e.g. amplitude
+        damping. This makes correcting for readout more difficult. In the simplest case, this
+        function runs the program normally ``(trials//2)`` times. The other half of the time,
+        it will insert an ``X`` gate prior to any ``MEASURE`` instruction and then flip the
+        measured classical bit back. Overall this has the effect of symmetrizing the readout error.
+
+        The details. Consider preparing the input bitstring |i> (in the computational basis) and
+        measuring in the Z basis. Then the Confusion matrix for the readout error is specified by
+        the probabilities
+
+            p_{j|i} := Pr(measured = j | prepared = i ).
+
+        In the case of a single qubit i,j \in [0,1] then:
+        there is no readout error if p_{0|0} = p_{1|1} = 1.
+        the readout error is symmetric if p_{0|0} = p_{1|1} = 1 - epsilon.
+        the readout error is asymmetric if p_{0|0} != p_{1|1}.
+
+        If your quantum computer has this kind of asymmetric readout error then
+        `qc.run_symmetrized_readout` will symmetrize the readout error.
+
+        The readout error above is only asymmetric on a single bit. In practice the confusion
+        matrix on n bits need not be symmetric, e.g. for two qubits p_{ij|ij} != 1 - epsilon for
+        all i,j. In these situations a more sophisticated means of symmetrization is needed; and
+        we use orthogonal arrays (OA) built from Hadamard matrices.
+
+        The symmetrization types are specified by an int; the types available are:
+        -1 -- exhaustive symmetrization uses every possible combination of flips
+         0 -- trivial that is no symmetrization
```

Comment:
so are we saying that this should behave identically to `run`?

Reply:
Not like `run`,  however `run_symmetrized_readout` should behave like `run_and_measure`.